src/build.ml

```diff
@@ -114,15 +115,24 @@ let lazy_no_targets t = Lazy_no_targets t
 let path p = Paths (Path.Set.singleton p)
 let paths ps = Paths (Path.Set.of_list ps)
 let path_set ps = Paths ps
-let paths_matching ~loc ~dir pred =
-  Paths_glob (ref (G_unevaluated (loc, dir, pred)))
+let paths_matching ~loc:_ ~dir pred = Paths_glob (dir, pred)
 let vpath vp = Vpath vp
 let dyn_paths t = Dyn_paths (t >>^ Path.Set.of_list)
 let dyn_path_set t = Dyn_paths t
+let dyn_deps t = Dyn_deps t
 let paths_for_rule ps = Paths_for_rule ps
 let universe = Universe
 let env_var s = Env_var s
 
+let of_deps (type a) (d : Dep.Set.t) : (a, a) t =
```

Comment:
Why not add a `Deps of Dep.Set.t` constructor here? If we already have a `Dep.Set.t` value, it feels weird to temporarily deconstruct it into a complex `Build.t` value just to reconstruct it into a `Dep.Set.t` later.

Reply:
Yeah, I agree with that. I did not want to make too many unrelated changes in this one PR. Will do that in a subsequent PR.